crates/evm/src/call.rs

```diff
@@ -80,7 +80,9 @@ impl<C: sov_modules_api::Context> Evm<C> {
 
         let mut cumulative_gas_used = 0;
         let mut log_index_start = 0;
-        if let Some(tx) = self.pending_transactions.last(working_set) {
+
+        if !self.pending_transactions.is_empty() {
```

Comment:
If they are not empty we are doing smth wrong

Reply:
remember when you were implementing sys txs and I asked you why you did some stuff in a certain way you said maybe we'll run sys txs at the end of the blocks in the future.

:)

Reply:
I was wrong =)

Reply:
I can make another pr that makes sure execute sys txs is called at the beginning. it will also save us some cycles

Reply:
Yes